components/centraldashboard/app/profile_controller.ts

```diff
@@ -0,0 +1,827 @@
+/**
```

Comment:
Let's name this file profile_controller_client to make it clear it's a client library and not part of our application server. It also might be useful to put it into a separate folder in case the generated Typescript doesn't comply with our tslint rules. When I tried building this, it failed because the linter was picking up things in the generated code that don't comply with the guidelines.

You can add a linterOptions section to the tslint.json file:
```
"linterOptions": {
        "exclude": [
            "app/clients"
        ]
    }
```

Reply:
Done! Doesn't make sense to rename now that it's in a folder called clients?